client/http-client/src/client.rs

```diff
@@ -160,11 +176,21 @@ impl ClientT for HttpClient {
 		let mut ordered_requests = Vec::with_capacity(batch.len());
 		let mut request_set = FxHashMap::with_capacity_and_hasher(batch.len(), Default::default());
 
-		let ids = self.id_manager.next_request_ids(batch.len())?;
+		let guard = self.id_manager.next_request_ids(batch.len())?;
+
+		let ids: Vec<Id> = guard
+			.inner()
+			.into_iter()
+			.map(|&id| match self.id_kind {
+				IdKind::Number => Id::Number(id),
+				IdKind::String => Id::Str(format!("{}", id).into()),
+			})
```

Comment:
It's a bit annoying to have to loop over and transform each id one by one. If `self.id_manager` knew what `IdKind` the user has configured, then the `next_request_ids()` call could returns the IDs of the right kind. Do you think such a refactoring is possible?